Controllers/FrontController.php

```diff
@@ -1,44 +1,116 @@
 <?php
+declare(strict_types=1);
 
 namespace Controllers;
 
+use Models\DataRegistry;
+use Models\Server;
+use Models\Session;
+use Models\Post;
+use Models\File;
+
+/**
+ * Main class for routing
+ *
+ * @package Controllers
+ */
 class FrontController
 {
-    protected $controller;
-    protected $action, $params;
-    public static $instance;
+    /**
+     * Given controller
+     */
+    protected string $controller;
+
+    /**
+     * Given action
+     */
+    protected string $action;
+
+    /**
+     * Given params values
+     */
+    protected ?array $params = null;
+
+    /**
+     * Request uri
+     */
+    protected string $request;
 
-    public static function getInstance()
+    /**
+     * Object for access to server data
+     */
+    private object $serverData;
+
+    public function __construct()
     {
-        if (!(self::$instance instanceof self)) {
-            self::$instance = new self();
-        }
-        return self::$instance;
+        $this->registerData();
+        $this->serverData = DataRegistry::getInstance()->get('server');
     }
 
-    private function __construct()
+    /**
+     * Register server and session models for encapsulating access
+     *
+     * @return void
+     * @throws \Exception
+     */
+    private function registerData(): void
     {
-        $request = $_SERVER['REQUEST_URI'];
-        $splits = explode('/', trim($request, '/'));
-        $this->controller = !empty($splits[0]) ?
-            $this->getFullControllerName(ucfirst($splits[0]) . 'Controller')
-            : $this->getFullControllerName('IndexController');
-        $this->action = !empty($splits[1]) ? $splits[1] . 'Action' : 'indexAction';
-        if (!empty($splits[2])) {
-            $keys = $values = [];
-            for ($i = 2, $count = count($splits); $i < $count; $i++) {
-                if ($i % 2 == 0) {
-                    $keys[] = $splits[$i];
-                } else {
-                    $values[] = $splits[$i];
+        $register = DataRegistry::getInstance();
+        $register->register('server', new Server\Manager());

```

Comment:
make **register** function to return $this instead of void and u can use a chain call here like 
```
register = DataRegistry::getInstance();
 $register->register('server', new Server\Manager())
->register('session', new Session\Manager())
->register('post', new Post\Manager())
->register('file', new File\Manager());
```